lemur/plugins/lemur_acme/plugin.py

```diff
@@ -715,3 +746,206 @@ def create_authority(options):
+            "helpMessage": "Certificate to use",
+        },
+        {
+            "name": "tokenDestination",
+            "type": "select",
+            "required": True,
+            "available": destination_list,
+            "helpMessage": "The destination to use to deploy the token.",
+        },
+    ]
+
+    def __init__(self, *args, **kwargs):
+        super(ACMEHttpIssuerPlugin, self).__init__(*args, **kwargs)
+
+        if len(self.destination_list) == 0:
+            destinations = destination_service.get_all()
+            for destination in destinations:
+                # we only want to use sftp destinations here
+                if destination.plugin_name == "sftp-destination":
+                    self.destination_list.append(destination.label)
+
+    def create_certificate(self, csr, issuer_options):
+        """
+        Creates an ACME certificate using the HTTP-01 challenge.
+
+        :param csr:
+        :param issuer_options:
+        :return: :raise Exception:
+        """
+        self.acme = AcmeHandler()
+        authority = issuer_options.get("authority")
+        create_immediately = issuer_options.get("create_immediately", False)
+        acme_client, registration = self.acme.setup_acme_client(authority)
+
+        orderr = acme_client.new_order(csr)
+
+        chall = []
+        for authz in orderr.authorizations:
+            # Choosing challenge.
+            # authz.body.challenges is a set of ChallengeBody objects.
+            for i in authz.body.challenges:
+                # Find the supported challenge.
+                if isinstance(i.chall, challenges.HTTP01):
+                    chall.append(i)
+
+        if len(chall) == 0:
+            raise Exception('HTTP-01 challenge was not offered by the CA server.')
+        else:
+            # Here we probably should create a pending certificate and make use of celery, but for now
+            # I'll ignore all of that
+            token_destination = None
+            for option in json.loads(issuer_options["authority"].options):
+                if option["name"] == "tokenDestination":
+                    token_destination = destination_service.get_by_label(option["value"])
+
+            if token_destination is None:
+                raise Exception('No token_destination configured for this authority. Cant complete HTTP-01 challenge')
+
+        destination_plugin = plugins.get(token_destination.plugin_name)
+
+        for challenge in chall:
+            response, validation = challenge.response_and_validation(acme_client.net.key)
+
+            destination_plugin.upload_acme_token(challenge.chall.path, validation, token_destination.options)
+
+            # Let the CA server know that we are ready for the challenge.
+            acme_client.answer_challenge(challenge, response)
+
+        current_app.logger.info("Uploaded HTTP-01 challenge tokens, trying to poll and finalize the order")
+
+        # It is possible to set a deadline time.
```

Comment:
shall we set a deadline? it defaults to 90s anyways.

Reply:
for the DNS path we are currently setting it to 6min:

https://github.com/Netflix/lemur/blob/983f9beacb85cb1ecb8ce10a5028d2664dffed23/lemur/plugins/lemur_acme/plugin.py#L179

Reply:
Right now I've set the timeout to 1minute, since we already handled the token upload. And unless there's an async `upload_acme_token` implementation (e.g. deploying a rule on a loadbalancer, that needs to be synced) we should be safe.

But I'll set it to the default 90s, and remove the comment. If there ever is an async upload implementation, it might be useful to make this configurable.